### PR TITLE
Makefile cleanup

### LINKED ISSUE #5: Cleanup Makefiles
- [x] Establish proper dependencies to enable parallelism (#11 )
- [x] Pass down variables to refer to `ROOT` instead relative addressing of includes and libraries
- [x] Since `make install` is performed for all submodules and are found in standard location we should not include the full path of the library and drop includes
    - [x] https://github.com/omec-project/c3po/blob/master/cdf/Makefile#L16-L18
    - [x] https://github.com/omec-project/c3po/blob/master/cdf/Makefile#L24-L25
    - [x] https://github.com/omec-project/c3po/blob/master/ctf/Makefile#L31-L34
    - [x] https://github.com/omec-project/c3po/blob/master/ctf/Makefile#L42-L45
    - [x] https://github.com/omec-project/c3po/blob/master/hss/Makefile#L27-L30
    - [x] https://github.com/omec-project/c3po/blob/master/hss/Makefile#L39-L43
    - [x] https://github.com/omec-project/c3po/blob/master/hssgtw/Makefile#L17-L20
    - [x] https://github.com/omec-project/c3po/blob/master/hssgtw/Makefile#L29-L32
    - [x] https://github.com/omec-project/c3po/blob/master/pcrf/Makefile#L17-L20
    - [x] https://github.com/omec-project/c3po/blob/master/pcrf/Makefile#L29-L31
    - [x] https://github.com/omec-project/c3po/blob/master/pcrf/tdf/Makefile#L16-L19
    - [x] https://github.com/omec-project/c3po/blob/master/pcrf/tdf/Makefile#L25-L29
    - [x] https://github.com/omec-project/c3po/blob/master/pcrf/tssf/Makefile#L16-L19
    - [x] https://github.com/omec-project/c3po/blob/master/pcrf/tssf/Makefile#L27-L29
    - [x] https://github.com/omec-project/c3po/blob/master/pcrf/pcef/Makefile#L16-L19
    - [x] https://github.com/omec-project/c3po/blob/master/pcrf/pcef/Makefile#L27-L29

    E.g.,
    ```
    diff --git a/cdf/Makefile b/cdf/Makefile
    index e7d23dc..2bde01c 100644
    --- a/cdf/Makefile
    +++ b/cdf/Makefile
    @@ -13,16 +13,14 @@ CFLAGS := -g -pthread -std=c++0x # -Wall
    LFLAGS := -g -pthread -lpthread
    LIBS := \
    ../util/lib/libc3po.a \
    - /usr/local/lib/libcares.so \
    - /usr/local/lib/libfdcore.so \
    - /usr/local/lib/libfdproto.so \
    + -lcares \
    + -lfdcore \
    + -lfdproto \
    -lmemcached \
    -lrt
    INCS := \
    -I./include \
    -I../util/include \
    - -I../modules/rapidjson/include \
    - -I /usr/local/include/freeDiameter
    ```

### PR DESCRIPTION
**Pull request by krsna1729**
Set dependencies to enable parallelism
Remove any reference to location of system installed libraries and
includes.
Use ROOT to convey top level dir to sub makes
Use WHAT to compile single component

Fixes: #5

Signed-off-by: Saikrishna Edupuganti <saikrishna.edupuganti@intel.com>

### CODE REVIEW

File: cdf/Makefile [hunk 1]
```diff
@@ -13,16 +13,14 @@ CFLAGS := -g -pthread -std=c++0x # -Wall
...
+ -lfdproto \
  -lmemcached \
  -lrt
 INCS := \
- -I./include \
- -I../util/include \
- -I../modules/rapidjson/include \
- -I /usr/local/include/freeDiameter
+ -I ./include \
+ -I ../util/include
```
Comment (suresh-marikkannu) [inline cdf/Makefile]: We can use `$ROOT` here as well.

Comment (krsna1729) [reply cdf/Makefile]: Fixed

**Review by somnathchakrabarti — APPROVED** [review]
OK to merge

**Comment by somnathchakrabarti** [comment]
Merged to master.

**Comment by krsna1729** [comment]
retest this please

**Comment by krsna1729** [comment]
retest this please

**Pull request merged**